frontend/src/routes/Infrastructure/VirtualMachines/VirtualMachinesPage.tsx

```diff
@@ -140,6 +148,17 @@ function VirtualMachineTable() {
     ]
   }, [searchResultItems, t])
 
+  const extensionColumns: IAcmTableColumn<ISearchResult>[] = useMemo(
+    // get the column extensions for the virtual machine
+    () => getVirtualMachineColumnExtensions(acmExtensions?.virtualMachineListColumn || []),
+    [acmExtensions]
+  )
```

Comment:
I don't think `useMemo` here is doing much good (ref of `acmExtensions` changes on each re-render, check the implementation of `useAcmExtension` hook), this causes `useMemo` to be evaluated again and again as well.

Reply:
The purpose of useMemo here is both to prevent changes in acmExtensions, Also, to ensure that the same IAcmTableColumn array reference is returned when the column list remains unchanged.

Reply:
If VirtualMachines re-renders, it will not create a new acmExtensions reference, because acmExtensions is part of the PluginContextProvider and is defined at a higher level in the component tree.
```
export default function VirtualMachinesPlugin() {
  return (
    <PluginContextProvider>
      <LoadPluginData>
        <ACMNotReadyWarning>
          <VirtualMachines />
        </ACMNotReadyWarning>
      </LoadPluginData>
    </PluginContextProvider>
  )
}

```

Reply:
the VirtualMachines is passed as a child to PluginContextProvider, but it is not taking any parameters.

Reply:
Each time components calling the hook `useAcmExtension` re-renders a new ref of `acmExtension` is created: https://github.com/stolostron/console/blob/main/frontend/src/plugin-extensions/handler.ts#L9

Reply:
VirtualMachine page is called as a of PlugContextProvider from here: https://github.com/stolostron/console/blob/main/frontend/src/components/PluginContextProvider.tsx#L66

Reply:
The ACM VirtualMachine List page does not actively watch for changes; instead, it is built on top of the Search API. It only queries the Search API when the page loads for the first time. It never renders again.